contributing_guidelines.md

```diff
@@ -0,0 +1,191 @@
+
+[Top](#top)
+
+##<a name="section3"></a>Language of the main repository
+The main repository language MUST be English.
+
+[Top](#top)
+
+##<a name="section4"></a>Repository organization
+Each agent MUST have a dedicated folder. Each folder MUST be prefixed with `cygnus-`. For instance:
+
+* `cygnus-ngsi`
+* `cygnus-twitter`
+
+Each folder MUST have, at least, the following subdirectories and files:
+
+* `src/` → functional code and unit tests
+* `doc/` → documentation in Markdown language
+* `docker/` → everything about deploying Cygnus be means of Docker
+* `test/` → acceptance tests, e2e tests, performance tests, others
+* `neore/` or `re/` → everything about installing Cygnus by means of a RPM
+* `conf/` → templates for configuration files required to run the agent
+* `pom.xml` → Maven’s Project Object Model
+
+A folder with common content named `cygnus-common` MUST exist. It will be a Maven project in charge of building a Cygnus common library that SHOULD be used by all the agents, enforcing the reusability of certain code.
+
+Every source file at any agent folder or `cygnus-common` MUST be under a Java package, following this format:
+
+    com.telefonica.iot.cygnus.<etc>.<etc>…
+
+Typically, these are the Java packages that SHOULD be used, but any other can be added:
+
+* `com.telefonica.iot.cygnus.sinks` → for new sink development
+* `com.telefonica.iot.cygnus.handlers` → for new handlers development
+* `com.telefonica.iot.cygnus.backends` → for new backends development
+* `com.telefonica.iot.cygnus.containers` → for new Json containers development
+* ...
+
+Any Java package MUST match a couple of paths like these ones:
+
+* `/<agent_name>/src/main/java/com/telefonica/iot/cygnus/<etc>/<etc>/…` → functional code
+* `/<agent_name>/src/test/java/com/telefonica/iot/cygnus/<etc>/<etc>/…` → unit tests
+
+As can be seen, despite the repository organization, from a Java perspective all the agents code is under `com.telefonica.iot.cygnus`.
+
+[Top](#top)
+
+##<a name="section5"></a>Backlog
+The <i>issues</i> section of the main repository MUST be used for tracking all the features, hardening, bugs and task to be implemented by every agent.
+
+The name of each issue MUST follow the following format:
+
+    [<agent name>] [feature|hardening|bug|task] <short description>
+
+Where <i>short description</i> MAY enclose other “[...]” sublevels. For instance:
+
+    [cygnus-ngsi] [hardening] [grouping rules] Precompile patterns from regexes
+
+Alternatively, labels for each agent and task type SHOULD be created.
+
+Every issue MUST have a description as detailed as the creator considers, but it MUST be enough to understand the purpose of the issue and to allow the community starts a discussion.
+
+Every issue SHOULD have an associated sprint/milestone among the ones in the <i>milestonres</i> section on the main repository.
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/943/commits/7a7e87176ef7b38c0d193fa949b7e468094f8274